llvm/lib/CodeGen/AtomicExpandPass.cpp

```diff
@@ -1684,24 +1654,19 @@ bool AtomicExpand::expandAtomicOpToLibcall(
   IRBuilder<> Builder(I);
   IRBuilder<> AllocaBuilder(&I->getFunction()->getEntryBlock().front());
 
-  bool UseSizedLibcall = IsCap || canUseSizedAtomicCall(Size, Alignment, DL);
-  if (DL.isFatPointer(PointerOperand->getType()) &&
-      !DL.isFatPointer(DL.getGlobalsAddressSpace())) {
-    LLVM_DEBUG(dbgs() << "Cannot expand atomic operation using capability "
-                         "pointer in hybrid mode";
-               I->dump(););
-    return false;
-  }
+  const bool IsCapValue = DL.isFatPointer(ValTy);
+  bool UseSizedLibcall =
+      IsCapValue || canUseSizedAtomicCall(Size, Alignment, DL);
+  bool PointerOperandIsCap = DL.isFatPointer(PointerOperand->getType());
```

Comment:
Would rather this be consistent with IsCapValue. Would IsCapPointer be too confusing (otherwise the alternative is to change IsCapValue to ValueOperandIsCap...).

Reply:
I'll go for the second option. More verbose but less confusing.